ratis-server/src/main/java/org/apache/ratis/server/impl/ReadIndexHeartbeats.java

```diff
@@ -101,7 +101,7 @@ boolean receive(LogAppender logAppender, AppendEntriesReplyProto proto,
       final HeartbeatAck reply = replies.computeIfAbsent(
           logAppender.getFollowerId(), key -> new HeartbeatAck(logAppender));
       if (reply.receive(proto)) {
-        if (hasMajority.test(id -> replies.get(id).isAcknowledged())) {
+        if (hasMajority.test(id -> replies.get(id) != null && replies.get(id).isAcknowledged())) {
```

Comment:
Let's use `Optional` to add an `isAcknowledged` method:
```java
    boolean isAcknowledged(RaftPeerId id) {
      return Optional.ofNullable(replies.get(id)).map(HeartbeatAck::isAcknowledged).isPresent();
    }
```

Reply:
done